scripts/orchestrators/iac.bicep.sh

```diff
@@ -210,13 +210,25 @@ export -f deploy
 destroy() {
     local environmentName=${1}
     local layerName=${2}
+    local location=${3}
+    
+    deployments=$(az deployment sub list --query "[?tag=='GeneratedBy=symphony']" --query "[?tag=='EnvironmentName=${environmentName}']" --query "[?tag=='LayerName=${layerName}']" --query "[?location=='${location}']" --output json |jq -c -r '.[].name')
+    exit_code=$?
+    
+    for deployment in ${deployments}; do
+        echo "Deleting deployment : ${deployment}"
+        az deployment sub delete --name "${deployment}"
+        exit_code=$?
+    done
 
-    resourceGroups=$(az group list --tag "GeneratedBy=symphony" --tag "EnvironmentName=${environmentName}" --tag "LayerName=${layerName}" --query [].name --output tsv)
-
+    resourceGroups=$(az group list --tag "GeneratedBy=symphony" --tag "EnvironmentName=${environmentName}" --tag "LayerName=${layerName}" --query "[?location=='${location}']" --output json |jq -c -r '.[].name')
     for resourceGroup in ${resourceGroups}; do
         _information "Destroying resource group: ${resourceGroup}"
         az group delete --resource-group "${resourceGroup}" --yes
+        exit_code=$?
```

Comment:
I suggest checking whether each command executed successfully or failed, instead of simply returning the last `exit_code`. like, set the `exit_code` variable to `0` if all commands succeed or `1` if _any_ command failed. this allows for determining if there was any failure among the executed commands.

Reply:
actually, most (if not all) of the scripts should be doing `set -e` to stop execution if something fails.
It has happened to me more than once that some of these scripts continue running and everything seems to be fine, only to go back to read the actual output and see that 500 lines before there was an error that made the whole result useless

Reply:
Also, as @polatengin said, at least it should do an OR of the results, so if there is any non-zero result, it should return non-zero exit code

Reply:
addressed